### DISCUSSION TITLE
How/where can I change domain name?

### DISCUSSION THREAD

**Discussion by tenzan** [gravitl/netmaker#2652] [Q&A]
Hi,

I installed netmaker from DO's market place.
After droplet installed, specifying custom domain didn't go well, so I decided to proceed with randomly generated domain name.
Now I have netmaker working and trying to change domain name to a custom one.
Where can I change it?
Will I have to run additional commands once domain name updated?

Thanks!

**Comment by mattkasun** [comment]
- log into DO droplet
- stop all containers `docker compose down`
- edit /root/netmaker.env
  - change NM_DOMAIN to your custom domain 
- edit /root/Caddyfile
  - delete all lines with ` tls /root/certs/fullchain.pem /root/certs/privkey.pem  `
- restart all containers `docker compose up -d`

**Reply by tenzan** [reply]
Thanks, but Caddyfile doesn't contain `tls /root/certs/fullchain.pem /root/certs/privkey.pem`
```

# Dashboard
https://dashboard.{$NM_DOMAIN} {
        # Apply basic security headers
        header {
                # Enable cross origin access to *.{$NM_DOMAIN}
                Access-Control-Allow-Origin *.{$NM_DOMAIN}
                # Enable HTTP Strict Transport Security (HSTS)
                Strict-Transport-Security "max-age=31536000;"
                # Enable cross-site filter (XSS) and tell browser to block detected attacks
                X-XSS-Protection "1; mode=block"
                # Disallow the site to be rendered within a frame on a foreign domain (clickjacking protection)
                X-Frame-Options "SAMEORIGIN"
                # Prevent search engines from indexing
                X-Robots-Tag "none"
                # Remove the server name
                -Server
        }

        reverse_proxy http://netmaker-ui
}

# API
https://api.{$NM_DOMAIN} {
        reverse_proxy http://netmaker:8081
}

# TURN
https://turn.{$NM_DOMAIN} {
        reverse_proxy host.docker.internal:3479
}

# TURN API
https://turnapi.{$NM_DOMAIN} {
        reverse_proxy http://host.docker.internal:8089
}

# MQ
wss://broker.{$NM_DOMAIN} {
        reverse_proxy ws://mq:8883 # For EMQX websockets use `reverse_proxy ws://mq:8083`
}
```

**Reply by mattkasun** [reply]
In that case you just have to update the NM_DOMAIN in the netmaker.env file and restart the containers.

**Reply by tenzan** [reply]
I did following:

1.  Put containers down

```
root@netmaker:~# docker-compose down
Removing netmaker-ui ... done
Removing coredns     ... done
Removing mq          ... done
Removing caddy       ... done
Removing turn        ... done
Removing netmaker    ... done
Removing network root_default
```
2.  I changed `NM_DOMAIN=netmaker.cms.icu`  (Random domain looked like `nm.<droplet-public-ip>.nip.io`)

3. Brought conatiners back online
```
root@netmaker:~# docker-compose up -d
Creating network "root_default" with the default driver
Creating netmaker ... done
Creating caddy       ... done
Creating turn     ... done
Creating mq          ... done
Creating coredns     ... done
Creating netmaker-ui ... done
```

Now when accessing https://dashboard.netmaker.cms.icu/ I see an error
```
Hmm. We’re having trouble finding that site.

We can’t connect to the server at dashboard.netmaker.cms.icu.

If you entered the right address, you can:

    Try again later
    Check your network connection
    Check that Firefox has permission to access the web (you might be connected but behind a firewall)
```

**Reply by tenzan** [reply]
Looks like I need to make additional DNS related settings... But no idea where...

**Reply by mattkasun** [reply]
`docker logs caddy` errors?

Did you create DNS entries for *.netmaker.cms.icu

**Comment by tenzan** [comment]
Hi,

I think we're having certificate issues.

`/var/log/syslog`:

```
Nov  3 09:57:11 netmaker kernel: [ 2251.112620] [UFW BLOCK] IN=eth0 OUT= MAC=e6:34:4d:cd:15:0d:fe:00:00:00:01:01:08:00 SRC=51.79.191.150 DST=159.223.94.35 LEN=40 TOS=0x00 PREC=0x00 TTL=244 ID=62024 DF PROTO=TCP SPT=40709 DPT=9856 WINDOW=512 RES=0x00 SYN URGP=0
```

`docker logs -f --until=10m caddy`:
```
{"level":"error","ts":1699005843.529814,"logger":"tls.obtain","msg":"will retry","error":"[api.netmaker.cms.icu] Obtain: [api.netmaker.cms.icu] solving challenge: api.netmaker.cms.icu: [api.netmaker.cms.icu] authorization failed: HTTP 0  -  (ca=https://acme.zerossl.com/v2/DV90)","attempt":7,"retrying_in":1200,"elapsed":2624.91751771,"max_duration":2592000}
```

**Reply by mattkasun** [reply]
api.netmaker.cms.icu does not resolve.... you have to create dns records.

**Reply by tenzan** [reply]
If not mistaken, `api.netmaker.cms.icu` will point to same IP address as `netmaker.cms.icu`.
I have created DNS records on Digital Ocean and now able to ping it, but accessing `https://dashboard.netmaker.cms.icu/` is showing same error on browser.
Looking at logs again I noticed another error. I'm not sure if it was there from the beginning or not.
```
Nov  4 04:02:54 netmaker netclient[1939]: [netclient] 2023-11-04 04:02:54 could not connect to broker at nm.159-223-94-35.nip.io
Nov  4 04:02:54 netmaker netclient[1939]: [netclient] 2023-11-04 04:02:54 error publishing checkin connection timeout
```

From the error above I'm under impression, just changing `NM_DOMAIN=` is not sufficient.

**Reply by tenzan** [reply]
It seems, I need DNS records for domains
turn.netmaker.cms.icu
broker.netmaker.cms.icu
turn.netmaker.cms.icu

Will create DNS records and see how it goes.
